pkg/apis/contrail/v1alpha1/swiftproxy_types.go

```diff
@@ -17,14 +17,15 @@ type SwiftProxySpec struct {
 // SwiftProxyConfiguration is the Spec for the keystone API.
 // +k8s:openapi-gen=true
 type SwiftProxyConfiguration struct {
-	MemcachedInstance     string       `json:"memcachedInstance,omitempty"`
-	ListenPort            int          `json:"listenPort,omitempty"`
-	KeystoneInstance      string       `json:"keystoneInstance,omitempty"`
-	KeystoneSecretName    string       `json:"keystoneSecretName,omitempty"`
-	CredentialsSecretName string       `json:"credentialsSecretName,omitempty"`
-	SwiftConfSecretName   string       `json:"swiftConfSecretName,omitempty"`
-	RingConfigMapName     string       `json:"ringConfigMapName,omitempty"`
-	Containers            []*Container `json:"containers,omitempty"`
+	MemcachedInstance     string             `json:"memcachedInstance,omitempty"`
+	ListenPort            int                `json:"listenPort,omitempty"`
+	KeystoneInstance      string             `json:"keystoneInstance,omitempty"`
+	KeystoneSecretName    string             `json:"keystoneSecretName,omitempty"`
+	CredentialsSecretName string             `json:"credentialsSecretName,omitempty"`
+	SwiftConfSecretName   string             `json:"swiftConfSecretName,omitempty"`
+	RingConfigMapName     string             `json:"ringConfigMapName,omitempty"`
+	Containers            []*Container       `json:"containers,omitempty"`
+	SwiftServiceType      corev1.ServiceType `json:"swiftServiceType,omitempty"`
```

Comment:
It would be nice to allow picking only valid service type and "", not any string, so then the validation of service type will be at api level, since right now i see that we can type in CR that the servicetype is "abc" and then the load balancer will be created. using kubebuilder you can create an enum which can have one value from the set e.g.: ClusterIP, NodePort, LoadBalancer, ExternalName, "". I think that would be a good feature